framework/CodeInterfaces/RAVEN/RAVENInterface.py

```diff
@@ -239,21 +239,30 @@ def createNewInput(self, currentInputFiles, oriInputFiles, samplerType, **Kwargs
       if convDict['scalar']:
         # call conversion, value changes happen in-place
         module.manipulateScalarSampledVariables(modifDict)
-
     # we work on batchSizes here
     newBatchSize = Kwargs['NumMPI']
     internalParallel = Kwargs.get('internalParallel',False)
     if int(Kwargs['numberNodes']) > 0:
       # we are in a distributed memory machine => we allocate a node file
       nodeFileToUse = os.path.join(Kwargs['BASE_WORKING_DIR'],"node_" +str(Kwargs['INDEX']))
-      if os.path.exists(nodeFileToUse):
-        modifDict['RunInfo|mode'           ] = 'mpi'
-        modifDict['RunInfo|mode|nodefile'  ] = nodeFileToUse
-      else:
-        raise IOError(self.printTag+' ERROR: The nodefile "'+str(nodeFileToUse)+'" does not exist!')
+      if not os.path.exists(nodeFileToUse):
+        if "PBS_NODEFILE" not in os.environ:
+          raise IOError(self.printTag+' ERROR: The nodefile "'+str(nodeFileToUse)+'" and PBS_NODEFILE enviroment var do not exist!')
+        else:
+          nodeFileToUse = os.environ["PBS_NODEFILE"]
```

Comment:
If I am understanding correctly (and this is confusing enough that I might not be) we are putting a value in PBS_NODEFILE in MPISimulationMode.py and using that to pass data to RAVENInterface.  This pretty much will break in anything but PBS.